sdks/python/apache_beam/yaml/yaml_mapping.py

```diff
@@ -842,6 +845,107 @@ def _AssignTimestamps(
                           ).with_input_types(T).with_output_types(T)
 
 
+class PaneInfoTuple(NamedTuple):
+  is_first: bool
+  is_last: bool
+  timing: str
+  index: int
+  nonspeculative_index: int
+
+  @classmethod
+  def from_pane_info(cls, pane_info):
+    return cls(
+        pane_info.is_first,
+        pane_info.is_last,
+        windowed_value.PaneInfoTiming.to_string(pane_info.timing),
+        pane_info.index,
+        pane_info.nonspeculative_index)
+
+
+_WINDOWING_INFO_TYPES = {
+    'timestamp': Timestamp,
```

Comment:
(optional, not sure if I like this more) - thoughts on `event_timestamp`? It might be a little more explicit.

We could even allow a user to specify `processing_timestamp` (resolves to `time.time()`) as a convenience.

Reply:
Interesting idea. 

I'm going to leave it as timestamp. Event timestamps are actually attached to the element, and windows are always in terms of this time. It's also a bit deceptive because if I try to inspect/reify processing time, that's not what will be used when anything else tries to act on it. (In addition, one needs triggers/timers to get into processing time and that's a whole can of worms we don't expose to YAML.)